sympy/functions/special/zeta_functions.py

```diff
@@ -214,6 +214,25 @@ def _eval_rewrite_as_zeta(self, z, s, a, **kwargs):
     def _eval_rewrite_as_polylog(self, z, s, a, **kwargs):
         return self._eval_rewrite_helper(polylog)
 
+    def _eval_is_finite(self):
+        z, s, a = self.args
+        if z.is_infinite or s.is_infinite or a.is_infinite:
+            # TODO: Write a valid implementation for infinite numbers.
+            return
+        if z == 0:
+            if a.is_zero is False:
+                return True
+            if s.is_real:
+                return s.is_nonpositive
+            return
+        if a.is_integer and a.is_nonpositive:
+            if s.is_real:
+                return s.is_nonpositive
+            return
+        exp_expr = self.expand(func=True)
+        if exp_expr != self:
+            return exp_expr.is_finite
```

Comment:
What is `expand(func=True)` being used for here?

We should avoid creating new expressions during an assumptions query. Just creating the new expression runs the risk of triggering an identical assumptions query which leads to infinite recursion.

Reply:
Ok, I did not think about that. My idea was that, since LerchPhi includes a lot of common functions as particular values (zeta function, polylog, Dirichlet eta, ...), I can try to write the function in terms of these other functions and then let them determine whether the function is finite or not. Is there a safe way to do it without the risks you are mentioning?

Reply:
There is not really a safe way to do it. We just have to accept that the core assumptions cannot resolve all queries:
https://docs.sympy.org/latest/guides/assumptions.html

Reply:
Ok, sorry for the delay, but I haven't had time these last few weeks. I understand then that this part of the code should be changed and all the checks need to be implemented from scratch. Is that correct?

Reply:
Yes, the checks need to be from scratch. It is also acceptable that some assumptions queries just return None.